src/gui/menu_species.cpp

```diff
@@ -341,6 +346,84 @@ void DissolveWindow::on_SpeciesSetChargesInSelectionAction_triggered(bool checke
     fullUpdate();
 }
 
+void DissolveWindow::on_SpeciesCopyChargesFromAtomTypesAction_triggered(bool checked)
+{
+    // Get the current Species (if a SpeciesTab is selected)
+    auto species = ui_.MainTabs->currentSpecies();
+    if (!species)
+        return;
+
+    if (QMessageBox::warning(this, "Copy Charges from Atom Types",
+                             "This will replace the species charges "
+                             "with those of the corresponding Atom Types.\n\n"
+                             "This cannot be undone! Proceed?",
+                             QMessageBox::StandardButton::Yes | QMessageBox::StandardButton::No,
+                             QMessageBox::StandardButton::No) == QMessageBox::StandardButton::Yes)
+    {
+        const auto atomTypes = dissolve().coreData().atomTypes();
+        for (auto &atom : species->atoms())
+        {
+            for (auto &atomType : atomTypes)
+            {
+                if (atomType == atom.atomType())
+                {
+                    atom.setCharge(atomType->charge());
+                    break;
+                }
+            }
+        }
+
+        setModified();
+        fullUpdate();
+    }
+}
+
+void DissolveWindow::on_SpeciesSetAtomTypeChargesFromSpeciesAction_triggered(bool checked)
+{
+    // Get the current Species (if a SpeciesTab is selected)
+    auto species = ui_.MainTabs->currentSpecies();
+    if (!species)
+        return;
+
+    std::map<std::shared_ptr<AtomType>, SampledDouble> charges;
+    auto atomTypes = dissolve().coreData().atomTypes();
+    for (auto &atomType : atomTypes)
+    {
+        for (auto &atom : species->atoms())
+        {
+            if (atom.atomType() == atomType)
+            {
+                charges[atomType] += atom.charge();
+            }
+        }
+    }
```

Comment:
I think you can just simplify this to:
```suggestion
    for (auto &atom : species->atoms())
        charges[atom.atomType()] += atom.charge();
```

Reply:
Yes, much cleaner!